### PR TITLE
nanocoap: fixed setting of option length

### PR DESCRIPTION
**Pull request by haukepetersen**
Options with a length > 12 byte did break the CoAP package. This PR fixes this and adapts the option length notation is described in section 3.1 in the RFC.

### CODE REVIEW

**Comment by haukepetersen** [comment]
@kaspar030: ping

File: nanocoap/nanocoap.c [hunk 1]
```diff
@@ -269,34 +269,39 @@ static uint32_t _decode_uint(uint8_t *pkt_pos, unsigned nbytes)
...
 
 size_t coap_put_option(uint8_t *buf, uint16_t lastonum, uint16_t onum, uint8_t *odata, size_t olen)
 {
     assert(lastonum <= onum);
-    int n = _put_odelta(buf, lastonum, onum, olen);
+
+    unsigned delta = (onum - lastonum);
+    *buf = 0;
+
+    unsigned n = _put_opt_len(buf, 1, 4, delta);
```
Comment (kaspar030) [inline nanocoap/nanocoap.c]: this is very confusing. ```_put_opt_len``` is used for both encoding the option length and the option delta, right?

Renaming the function is probably a good idea. At least there need to be proper comments.

File: nanocoap/nanocoap.c [hunk 2]
```diff
@@ -269,34 +269,39 @@ static uint32_t _decode_uint(uint8_t *pkt_pos, unsigned nbytes)
...
+    return offset;
 }
 
 size_t coap_put_option(uint8_t *buf, uint16_t lastonum, uint16_t onum, uint8_t *odata, size_t olen)
 {
     assert(lastonum <= onum);
-    int n = _put_odelta(buf, lastonum, onum, olen);
+
+    unsigned delta = (onum - lastonum);
+    *buf = 0;
```
Comment (kaspar030) [inline nanocoap/nanocoap.c]: why dis this needed?

Comment (haukepetersen) [inline nanocoap/nanocoap.c]: because the `_put_opt_len` function does not clear the length field anymore itself.

**Comment by haukepetersen** [comment]
better?

**Comment by kaspar030** [comment]
> better?

yes! please squash.

**Comment by haukepetersen** [comment]
rebased and squashed

**Comment by haukepetersen** [comment]
@kaspar030: are you ok to merge this now?

**Comment by haukepetersen** [comment]
@ping @ping @ping

**Pull request merged**